File: vstest/week2/2F-applegame/2828.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

#define DEBUG 0

int main(){
    // int N = 5;
    // int M = 2;
    // int J = 3;
    // vector<int> P = {1, 5, 3};

    int N, M, J;
    cin >> N >> M;
    cin >> J;
    vector<int> P;
    for (int i=0; i<J; i++){
        int tmp;
        cin >> tmp;
        P.push_back(tmp);
    }

    int width_M = M;
    int pos_M = 1;
    auto area_M = [](int p, int m){
        vector <int> area;
        for (int i = p; i < p+m; i++){
            area.push_back(i);
        }
        return area;
    };
    int valid_Pos = N - M + 1;

    if (DEBUG) cout << "\n";
    if (DEBUG) {
        cout << "( ";
        for (auto el : area_M(1,M))
            cout << el << " ";
        cout << ") ";
        cout << valid_Pos << "\n";

        cout << "( ";
        for (auto el : area_M(2,M))
            cout << el << " ";
        cout << ") ";
        cout << valid_Pos << "\n";
    }

    int ret = 0;
    if (valid_Pos == 1){
        cout << ret;
    } else {
        for (auto p : P){
            auto current_M = area_M(pos_M, M);
            if (find(current_M.begin(), current_M.end(), p) != current_M.end()){

            } else {
                if (p > current_M.back()) {
                    int dist = p - current_M.back();
                    pos_M += dist;
                    ret += dist;
                } else if (p < current_M.front()) {
                    int dist = p - current_M.front();
                    pos_M += dist;
                    ret += (-1 * dist);
                }
            }
        }
    }
    cout << ret;
    return 0;
}

File: vstest/week2/2F-applegame/2828.py
# N = 5
# M = 1
# J = 3

# P = [1, 5, 3]

N, M = [int(el) for el in input().split()]
J = int(input())
P = []
for idx in range(J):
    P.append(int(input()))
# 0|1|2|3|4|5|
#   M        
#   M_M     

width_M = M
pos_M = 1
area_M = lambda p, m: list(range(p, p + m))
valid_Pos = N - M + 1

# print(area_M(1,M), valid_Pos)
# print(area_M(2,M), valid_Pos)
# print(area_M(3,M), valid_Pos)

ret = 0
if valid_Pos == 1:
    print(ret)
    pass
else:
    for p in P:
        current_M = area_M(pos_M,M)
        if p in current_M:
            # print(f"p:{p}, pos_M: {pos_M}, ret: {ret}")
            pass
        else:
            if p > current_M[-1]:
                dist = p - current_M[-1]
                pos_M = pos_M + dist
                ret += dist
            elif p < current_M[0]:
                dist = p - current_M[0]
                pos_M = pos_M + dist
                ret += (-1*dist)
            # print(f"p:{p}, pos_M: {pos_M}, ret: {ret}")
            # print(ret)
print(ret)
